### DISCUSSION TITLE
fuzzy matching for search results via fuse.js

### DISCUSSION THREAD

**Discussion by karaliunas** [orchidjs/tom-select#482] [Ideas]
I am trying to implement a custom plugin to integrate fuse.js lib to enable approximate string matching. It would be a useful plugin for some e-commerce cases when setting up a dedicated backend (ElasticSearch or Algolia) would simply be too much just to handle search.

I can get a query string and bind to load hook to get fuzzy matching results data from fuse.js in the console. It has a score property for each match so I think it would be possible to use it for sorting.

`self.on('type', (str: string) => {
		querystring = str;
}),`

`self.on('load', (items: []) => {
		if (querystring.length === 0) {
			return;
		}
		const fuse = new Fuse(items, options);
		const result = fuse.search(querystring);
		console.log(result);
});`

Now I am stuck because I don't know how to pass those results to `refreshOptions` method (it is there where the logic of finding results is implemented). What I think we need is an additional hook to augment currently internal `search(query:string)` method. Maybe creating a hook in search method something like this: 
`if (trigger) self.trigger('search', queryString);` would be enough?  I would be willing to create a PR for that. Thanks!

**Comment by oyejorge** [comment]
Maybe something like this:
```js
class FuzzyTom extends TomSelect{

   // override search method
   search(query){
     let results = fuse.search(querystring);
     //...
     return results;
  }
}

**Reply by karaliunas** [reply]
thanks for a great solution @oyejorge! Will try and let you know how it goes

**Comment by leeoniya** [comment]
or not Fuse

https://github.com/leeoniya/uFuzzy#performance
